x-pack/packages/kbn-entities-schema/src/schema/common.ts

```diff
@@ -44,20 +44,21 @@ export const docCountMetricSchema = z.object({
   filter: filterSchema,
 });
 
-export const durationSchema = z
-  .string()
-  .regex(/^\d+[m|d|s|h]$/)
-  .transform((val: string) => {
-    const parts = val.match(/(\d+)([m|s|h|d])/);
-    if (parts === null) {
-      throw new Error('Unable to parse duration');
-    }
-    const value = parseInt(parts[1], 10);
-    const unit = parts[2] as 'm' | 's' | 'h' | 'd';
-    const duration = moment.duration(value, unit);
-    duration.toJSON = () => val;
-    return duration;
```

Comment:
updated this schema to only use moment for validation and return the raw string since we only use the literal value in downstream code. Feels like if client code needs to do more complex operation on the type they can use their own logic or 3p library so we don't have to tie the schema to a specific (and custom) type